Patch by E5ten.

diff --git a/libelf/gelf_symshndx.c b/libelf/gelf_symshndx.c
index 81789521..7ba9f0e1 100644
--- a/libelf/gelf_symshndx.c
+++ b/libelf/gelf_symshndx.c
@@ -48,6 +48,12 @@ gelf_getsymshndx(Elf_Data *d, Elf_Data *id, int ndx, GElf_Sym *dst,
 	if (gelf_getsym(d, ndx, dst) == 0)
 		return (NULL);
 
+	if (lid == NULL) {
+		if (shindex)
+			*shindex = 0;
+		return (dst);
+	}
+
 	if (lid == NULL || (scn = lid->d_scn) == NULL ||
 	    (e = scn->s_elf) == NULL || (e != ld->d_scn->s_elf) ||
 	    shindex == NULL) {
