All credit goes to E5ten.
---
diff --git a/meson.build.old b/meson.build
index 932eb13..b192b51 100644
--- a/meson.build
+++ b/meson.build
@@ -1706,7 +1706,7 @@ if host_machine.system() == 'windows'
     prog_bison = find_program('bison', 'yacc', required : with_any_opengl)
   endif
 else
-  prog_bison = find_program('bison', required : with_any_opengl)
+  prog_bison = find_program('bison', 'byacc', required : with_any_opengl)
 
   # Disable deprecated keyword warnings, since we have to use them for
   # old-bison compat.  See discussion in
diff --git a/src/compiler/glsl/glcpp/glcpp-lex.l b/src/compiler/glsl/glcpp/glcpp-lex.l
index e07739b657c..bb4aee3c948 100644
--- a/src/compiler/glsl/glcpp/glcpp-lex.l
+++ b/src/compiler/glsl/glcpp/glcpp-lex.l
@@ -27,7 +27,7 @@
 #include <ctype.h>
 
 #include "glcpp.h"
-#include "glcpp-parse.h"
+#include "glcpp-parse.tab.h"
 
 /* Flex annoyingly generates some functions without making them
  * static. Let's declare them here. */
diff --git a/src/compiler/glsl/glcpp/meson.build b/src/compiler/glsl/glcpp/meson.build
index 9fc8d9d0286..45dfb170145 100644
--- a/src/compiler/glsl/glcpp/meson.build
+++ b/src/compiler/glsl/glcpp/meson.build
@@ -19,12 +19,11 @@
 # SOFTWARE.
 
 glcpp_parse = custom_target(
-  'glcpp-parse.[ch]',
+  'glcpp-parse.tab.[ch]',
   input : 'glcpp-parse.y',
-  output : ['glcpp-parse.c', 'glcpp-parse.h'],
+  output : ['glcpp-parse.tab.c', 'glcpp-parse.tab.h'],
   command : [
-    prog_bison, '-o', '@OUTPUT0@', '-p', 'glcpp_parser_',
-    '--defines=@OUTPUT1@', '@INPUT@',
+    prog_bison, '-p', 'glcpp_parser_', '-d', '-b', '@OUTDIR@/@BASENAME@', '@INPUT@',
   ],
 )
 
diff --git a/src/compiler/glsl/glsl_lexer.ll b/src/compiler/glsl/glsl_lexer.ll
index 7d7ee0c00ff..4d76260d231 100644
--- a/src/compiler/glsl/glsl_lexer.ll
+++ b/src/compiler/glsl/glsl_lexer.ll
@@ -26,7 +26,7 @@
 #include "util/strtod.h"
 #include "ast.h"
 #include "glsl_parser_extras.h"
-#include "glsl_parser.h"
+#include "glsl_parser.tab.h"
 #include "main/mtypes.h"
 
 static int classify_identifier(struct _mesa_glsl_parse_state *, const char *,
diff --git a/src/compiler/glsl/glsl_parser_extras.cpp b/src/compiler/glsl/glsl_parser_extras.cpp
index e9ef5b56d8b..aaf4843638b 100644
--- a/src/compiler/glsl/glsl_parser_extras.cpp
+++ b/src/compiler/glsl/glsl_parser_extras.cpp
@@ -36,7 +36,7 @@
 #include "util/mesa-sha1.h"
 #include "ast.h"
 #include "glsl_parser_extras.h"
-#include "glsl_parser.h"
+#include "glsl_parser.tab.h"
 #include "ir_optimization.h"
 #include "loop_analysis.h"
 #include "builtin_functions.h"
diff --git a/src/compiler/glsl/meson.build b/src/compiler/glsl/meson.build
index 74cb8973077..7dfd433d78c 100644
--- a/src/compiler/glsl/meson.build
+++ b/src/compiler/glsl/meson.build
@@ -23,10 +23,10 @@ subdir('glcpp')
 glsl_parser = custom_target(
   'glsl_parser',
   input : 'glsl_parser.yy',
-  output : ['glsl_parser.cpp', 'glsl_parser.h'],
+  output : ['glsl_parser.tab.cpp', 'glsl_parser.tab.h'],
   command : [
-    prog_bison, '-o', '@OUTPUT0@', '-p', '_mesa_glsl_', '--defines=@OUTPUT1@',
-    '@INPUT@',
+    prog_bison, '-p', '_mesa_glsl_', '-b', '@OUTDIR@/@BASENAME@', '-d',
+    '-o', '@OUTPUT0@', '@INPUT@',
   ],
 )
 
diff --git a/src/mesa/program/meson.build b/src/mesa/program/meson.build
index 5f3325ff53c..236e6049d42 100644
--- a/src/mesa/program/meson.build
+++ b/src/mesa/program/meson.build
@@ -29,5 +29,5 @@ program_parse_tab = custom_target(
   'program_parse_tab.[ch]',
   input : 'program_parse.y',
   output : ['program_parse.tab.c', 'program_parse.tab.h'],
-  command : [prog_bison, '-o', '@OUTPUT0@', '--defines=@OUTPUT1@', '@INPUT@'],
+  command : [prog_bison, '-b', '@OUTDIR@/@BASENAME@', '-d', '@INPUT@'],
 )
