diff --git a/Makefile.in b/Makefile.in
index c493f5e..e3c432b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -22,10 +22,10 @@ objs = carg_parser.o main.o
 all : $(progname_static) $(progname_shared)
 
 lib$(libname).a : lzlib.o
-	$(AR) -rcs $@ $<
+	$(AR) -rcs $@ lzlib.o
 
 lib$(libname).so.$(pkgversion) : lzlib_sh.o
-	$(CC) $(LDFLAGS) $(CFLAGS) -fpic -fPIC -shared -Wl,--soname=lib$(libname).so.$(soversion) -o $@ $<
+	$(CC) $(LDFLAGS) $(CFLAGS) -fpic -fPIC -shared -Wl,--soname=lib$(libname).so.$(soversion) -o $@ lzlib_sh.o
 
 $(progname) : $(objs) lib$(libname).a
 	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $(objs) lib$(libname).a
@@ -40,13 +40,13 @@ lzcheck : lzcheck.o lib$(libname).a
 	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ lzcheck.o lib$(libname).a
 
 main.o : main.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) -DPROGVERSION=\"$(pkgversion)\" -c -o $@ $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) -DPROGVERSION=\"$(pkgversion)\" -c -o $@ main.c
 
 lzlib_sh.o : lzlib.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) -fpic -fPIC -c -o $@ $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) -fpic -fPIC -c -o $@ lzlib.c
 
-%.o : %.c
+.c.o:
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
 
 lzdeps = lzlib.h lzip.h cbuffer.c decoder.h decoder.c encoder_base.h \
          encoder_base.c encoder.h encoder.c fast_encoder.h fast_encoder.c
