diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c b/media/transport/webrtc/third_party/nICEr/src/stun/addrs-netlink.c
index ddecc34..b3a2400 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
@@ -72,7 +72,7 @@ static const int kMaxReadSize = 4096;
 
 static void set_ifname(nr_local_addr *addr, struct ifaddrmsg* msg) {
   assert(sizeof(addr->addr.ifname) > IF_NAMESIZE);
-  if_indextoname(msg->ifa_index, addr->addr.ifname);
+  //if_indextoname(msg->ifa_index, addr->addr.ifname);
 }
 
 static int get_siocgifflags(nr_local_addr *addr) {
