--- rustc-1.47.0-src/library/unwind/src/lib.rs
+++ rustc-1.47.0-src/library/unwind/src/lib.rs
@@ -21,8 +21,7 @@
 }
 
 #[cfg(target_env = "musl")]
-#[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
-#[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
+#[link(name = "unwind")]
 extern "C" {}
 
 #[cfg(target_os = "redox")]
--- rustc-1.47.0-src/library/unwind/build.rs
+++ rustc-1.47.0-src/library/unwind/build.rs
@@ -105,15 +105,6 @@
             cfg.file(root.join("src").join(src));
         }
 
-        if target_env == "musl" {
-            // use the same C compiler command to compile C++ code so we do not need to setup the
-            // C++ compiler env variables on the builders
-            cfg.cpp(false);
-            // linking for musl is handled in lib.rs
-            cfg.cargo_metadata(false);
-            println!("cargo:rustc-link-search=native={}", env::var("OUT_DIR").unwrap());
-        }
-
         cfg.compile("unwind");
     }
 }
